### PR TITLE
Github Actions to check osx config

### PR DESCRIPTION
**Pull request by sofianehaddad**
Travis is limited to 50 mn and build + cppcheck seem to require more
Thus this last one will be ignored or used for other specific check

Now we will use this continuous integration platform to check osx
configuration, including both cxx/py tests (without doc checking)

Here we are trying to use all dependencies. For the moment:
 - dlib (from brew) has compilation issues (bad cxx flags/includes)
 - bonmin is not used as there is no brew formula and requires some work
 to integrate it.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+on: push
+jobs:
+  build:
+    runs-on: macos-latest
+    env:
+      LDFLAGS: -L/usr/local/opt/openblas/lib
+      CPPFLAGS: -I/usr/local/opt/openblas/include
```
Comment (jschueller) [inline .github/workflows/build.yml:7]: you can set MAKEFLAGS onc in env instead of calling sysctl in each command to set the number of jobs

Comment (jschueller) [reply .github/workflows/build.yml:7]: maybe then repush to retrigger as I've just enabled actions in PRs from forks in openturns settings

Comment (sofianehaddad) [reply .github/workflows/build.yml:7]: indeed

Comment (jschueller) [reply .github/workflows/build.yml:7]: or maybe the yml must already be present, like what we had for the sphinx artifacts

**Comment by jschueller** [comment]
you can drop the travis config file then

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    steps:
+    - uses: actions/checkout@v2
+    - run: brew upgrade
+    - run: brew install openblas swig boost python3 tbb nlopt cminpack ceres-solver bison flex hdf5 ipopt r
```
Comment (jschueller) [inline .github/workflows/build.yml:11]: did you try mpfr/mpc ?

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    - run: cd $HOME/work/openturns && git clone https://github.com/yixuan/spectra.git spectra
+    - run: cd $HOME/work/openturns/spectra && git checkout v0.9.0
+    - run: cd $HOME/work/openturns/spectra && cmake -DBUILD_TESTS=TRUE -DCMAKE_INSTALL_PREFIX=$HOME/ .
+    - run: cd $HOME/work/openturns/spectra && make all && make install
```
Comment (jschueller) [inline .github/workflows/build.yml:15]: with cmake you can do "make install" without need for "make all"

File: .github/workflows/build.yml (hunk 2 above)
Comment (sofianehaddad) [reply .github/workflows/build.yml:11]: Have a look at https://github.com/sofianehaddad/openturns/runs/2341501238?check_suite_focus=true
Both `MPFR`, `MPC` are available

Comment (jschueller) [reply .github/workflows/build.yml:11]: ah ok

**Comment by jschueller** [comment]
... and the travis link in the README

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    - run: cd $HOME/work/openturns/spectra && git checkout v0.9.0
+    - run: cd $HOME/work/openturns/spectra && cmake -DBUILD_TESTS=TRUE -DCMAKE_INSTALL_PREFIX=$HOME/ .
+    - run: cd $HOME/work/openturns/spectra && make install ${MAKEFLAGS}
+    - run: cd $HOME/work/openturns && git clone https://github.com/jeromerobert/hmat-oss.git hmat-oss
+    - run: cd $HOME/work/openturns/hmat-oss/ && git checkout 1.6.1
+    - run: cd $HOME/work/openturns/hmat-oss/ && cmake -DBUILD_EXAMPLES=OFF -DCBLAS_INCLUDE_DIR=/usr/local/opt/openblas/include -DCMAKE_MACOSX_RPATH=ON .
+    - run: cd $HOME/work/openturns/hmat-oss/ && make install ${MAKEFLAGS}
+    - run: pip3 install matplotlib scipy chaospy pandas dill
+    - run: cmake -DR_EXECUTABLE="dummyR" -DPYTHON_EXECUTABLE=/usr/local/bin/python3 -DFLEX_EXECUTABLE=/usr/local/opt/flex/bin/flex -DBISON_EXECUTABLE=/usr/local/opt/bison/bin/bison -DSPECTRA_INCLUDE_DIR=/Users/runner/include .
+    - run: make ${MAKEFLAGS}
```
Comment (jschueller) [inline .github/workflows/build.yml:23]: no need to specify ${MAKEFLAGS} for make commands

File: README.rst [hunk 5]
```diff
@@ -1,9 +1,6 @@
 .. image:: https://circleci.com/gh/openturns/openturns.svg?style=shield
     :target: https://circleci.com/gh/openturns/openturns
 
-.. image:: https://travis-ci.org/openturns/openturns.svg?branch=master
-    :target: https://travis-ci.org/openturns/openturns
-
```
Comment (jschueller) [inline README.rst:6]: also we mention travis in the dev doc

File: .github/workflows/build.yml [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+      LDFLAGS: -L/usr/local/opt/openblas/lib
+      CPPFLAGS: -I/usr/local/opt/openblas/include
+      MAKEFLAGS: "-j $(sysctl -n hw.logicalcpu)"
+    steps:
+    - uses: actions/checkout@v2
+    - run: brew upgrade
+    - run: brew install openblas swig boost python3 tbb nlopt cminpack ceres-solver bison flex hdf5 ipopt r
+    - run: cd $HOME/work/openturns && git clone https://github.com/yixuan/spectra.git spectra
+    - run: cd $HOME/work/openturns/spectra && git checkout v0.9.0
+    - run: cd $HOME/work/openturns/spectra && cmake -DBUILD_TESTS=TRUE -DCMAKE_INSTALL_PREFIX=$HOME/ .
```
Comment (jschueller) [inline .github/workflows/build.yml:15]: seems you can write a long command in a single run instruction, with | and \ to skip lines if needed

File: .github/workflows/build.yml [hunk 7]
```diff
@@ -0,0 +1,31 @@
+on: push
```
Comment (jschueller) [inline .github/workflows/build.yml:1]: seems it must be explicitely enabled for pull requests:
"pull_request:"
at least that's what I see in hmat

**Comment by jschueller** [comment]
I think you should enable unity builds to speed up the compilation, see run_docker_linux

File: .github/workflows/build.yml (hunk 7 above)
Comment (sofianehaddad) [reply .github/workflows/build.yml]: yes but we might have both (`pull_request` +` push`)
However I am wondering if there is some jobs do not run concurrently in that case. I can add specific push on `master`

File: .github/workflows/build.yml (hunk 6 above)
Comment (sofianehaddad) [reply .github/workflows/build.yml]: Done

File: README.rst (hunk 5 above)
Comment (sofianehaddad) [reply README.rst:6]: Done

File: .github/workflows/build.yml (hunk 4 above)
Comment (sofianehaddad) [reply .github/workflows/build.yml]: seem that `makeflags` is generating some troubles

**Comment by sofianehaddad** [comment]
@jschueller Some cosmetics are needed but the main important is that some tests are failing.
Any idea on how to solve https://github.com/openturns/openturns/runs/2352281215?check_suite_focus=true#step:12:478 and https://github.com/openturns/openturns/runs/2352281215?check_suite_focus=true#step:12:1655

I can not reproduce on my `osx` laptop

**Comment by jschueller** [comment]
the generalizedpareto can be skipped (ctest -E) now and debugged later
cppcheck_EmpiricalBernsteinCopula_std can be fixed (only a digit change),

**Comment by jschueller** [comment]
it was "ctest -E GeneralizedParetoFactory_std", my bad

**Comment by jschueller** [comment]
cxx tests take forever: 1h for 50%

**Comment by sofianehaddad** [comment]
strange. Platform sometimes hangs and I do not know why

File: .github/workflows/build.yml [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+           make install -j$(sysctl -n hw.logicalcpu)
+    - run:  |
+        cmake -DR_EXECUTABLE="dummyR" -DPYTHON_EXECUTABLE=/usr/local/bin/python3 \
+        -DFLEX_EXECUTABLE=/usr/local/opt/flex/bin/flex \
+        -DBISON_EXECUTABLE=/usr/local/opt/bison/bin/bison \
+        -DSPECTRA_INCLUDE_DIR=/Users/runner/include \
+        -DCMAKE_UNITY_BUILD=ON -DCMAKE_UNITY_BUILD_BATCH_SIZE=32 .
+    - run: make install -j$(sysctl -n hw.logicalcpu)
+    - run: make tests -j$(sysctl -n hw.logicalcpu)
+    - run: ctest -E GeneralizedParetoFactory_std --output-on-failure --timeout 100
```
Comment (jschueller) [inline .github/workflows/build.yml:36]: the tests dont run in parallel anymore ?

Comment (sofianehaddad) [reply .github/workflows/build.yml:36]: indeed
I will fix it

**Comment by sofianehaddad** [comment]
@jschueller seem that all troubles have been fixed

**Review by jschueller — APPROVED** [review]

**Comment by jschueller** [comment]
i'll let it run one last time and merge it

**Comment by sofianehaddad** [comment]
ok. Maybe we can try later to seem how it works with `ubuntu` platform?
For info, last `osx` run (without doc of course) took 1h (install of dep + compilation of spectra + hmat + openturns + ctest)
It might be interesting to benchmark for `ubuntu`

**Comment by jschueller** [comment]
why not

**Pull request merged**